### PR TITLE
feat: integration with sentry

### LINKED ISSUE #6164: Feature: integration with sentry
**Summary**

We can report backtrace to sentry for better error tracking.

ref: https://github.com/getsentry/sentry-rust/tree/master/sentry-tracing

### PR DESCRIPTION
**Pull request by PsiACE**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

**usage**

- self host sentry or use SaaS
- create a new project and get DSN
- fill environment and run databend

sentry-databend

## Changelog

- New Feature

## Related Issues

Fixes #6164

### CODE REVIEW

**Comment by Xuanwo** [comment]
Is it possible to integrate with sentry by open tracing?

**Comment by Xuanwo** [comment]
We need something like https://pkg.go.dev/github.com/open-telemetry/opentelemetry-collector-contrib/exporter/sentryexporter#section-readme

**Comment by PsiACE** [comment]
> Is it possible to integrate with sentry by open tracing?

I think it's possible, but I don't see this in practice at the moment.

> It doesn't make sense for applications to integrate tracing vendors one by one...

You're right. I will explore other, simpler integration options.

**Review by everpcpc — APPROVED** [review]

File: query/bin/databend-query.rs [hunk 1]
```diff
@@ -52,6 +53,16 @@ async fn main(_global_tracker: Arc<RuntimeTracker>) -> common_exception::Result<
         "databend-query-{}@{}:{}",
         conf.query.cluster_id, conf.query.mysql_handler_host, conf.query.mysql_handler_port
     );
+
+    let mut _sentry_guard = None;
+    let bend_sentry_env = env::var("DATABEND_SENTRY").unwrap_or_else(|_| "".to_string());
+    if !bend_sentry_env.is_empty() {
+        _sentry_guard = Some(sentry::init((bend_sentry_env, sentry::ClientOptions {
+            release: sentry::release_name!(),
+            ..Default::default()
```
[the remark this answers is not in the document]
Comment (PsiACE) [reply query/bin/databend-query.rs:64]: add `databend_semver` macro

File: common/tracing/src/logging.rs [hunk 2]
```diff
@@ -104,6 +105,16 @@ pub fn init_global_tracing(
         jaeger_layer = Some(tracing_opentelemetry::layer().with_tracer(tracer));
     }
 
+    // Sentry Layer.
+    let mut sentry_layer = None;
+    let bend_sentry_env = env::var("DATABEND_SENTRY").unwrap_or_else(|_| "".to_string());
+    if !bend_sentry_env.is_empty() {
+        sentry_layer = Some(sentry_tracing::layer().event_filter(|md| match md.level() {
+            &tracing::Level::ERROR => EventFilter::Event,
+            _ => EventFilter::Ignore,
```
[the remark this answers is not in the document]
Comment (PsiACE) [reply common/tracing/src/logging.rs]: `&tracing::Level::ERROR | &tracing::Level::WARN`

File: metasrv/src/bin/metasrv.rs [hunk 3]
```diff
@@ -39,6 +40,15 @@ async fn main(_global_tracker: Arc<RuntimeTracker>) -> common_exception::Result<
         return Ok(());
     }
 
+    let mut _sentry_guard = None;
+    let bend_sentry_env = env::var("DATABEND_SENTRY").unwrap_or_else(|_| "".to_string());
+    if !bend_sentry_env.is_empty() {
+        _sentry_guard = Some(sentry::init((bend_sentry_env, sentry::ClientOptions {
+            release: sentry::release_name!(),
```
[the remark this answers is not in the document]
Comment (PsiACE) [reply metasrv/src/bin/metasrv.rs]: add `databend_semver` macro

File: query/bin/databend-query.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (PsiACE) [reply query/bin/databend-query.rs:64]: I think a simple configuration would be basically usable. Skip it for now.

**Comment by PsiACE** [comment]
It looks like the Sentry Performance has some features similar to the Jaeger.

I think we can add an environment variable to enable the sentry tracing feature so that we can make better use of sentry.

**Comment by PsiACE** [comment]
Screenshot 2022-06-25 at 15-28-53 Performance - Event Details - sentry - Sentry

**Comment by BohuTANG** [comment]
Please add some doc on how to profile databend using sentry.

**Comment by PsiACE** [comment]
> Please add some doc on how to profile databend using sentry.

I feel it is better suited to tracing, now that the documentation has been added. 

view doc: https://databend-git-fork-psiace-sentry-databend.vercel.app/doc/faq/how-to-tracing#error-tracking-and-performance-monitoring-with-sentry

File: docs/doc/100-faq/20-how-to-tracing.md [hunk 4]
```diff
@@ -181,7 +181,49 @@ Open http://127.0.0.1:16686/
...
+  ```
+
+  <img src="/img/tracing/sentry-error.png"/>
+
+- **Also Enable Performance Monitoring**
+
+  Setting `SENTRY_TRACES_SAMPLE_RATE` greater than `0.0` will allow sentry to perform trace sampling, which will help set up performance monitoring.
+
+  ```bash
+  DATABEND_SENTRY_DSN="<your-sentry-dsn>" SENTRY_TRACES_SAMPLE_RATE=1.0 LOG_LEVEL=DEBUG ./databend-query
```
Comment (BohuTANG) [inline docs/doc/100-faq/20-how-to-tracing.md:219]: log level is TRACE is ok?

Comment (PsiACE) [reply docs/doc/100-faq/20-how-to-tracing.md:219]: It is also possible, but debug is enough

Comment (PsiACE) [reply docs/doc/100-faq/20-how-to-tracing.md:219]: I think it's ok to be consistent with jaeger's log_level.

**Review by dantengsky — APPROVED** [review]
LGTM

**Pull request merged**